projects/ngx-d3timeline/src/lib/content/drag.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { EventRectangleDragEvent } from './event-rectangle-drag-event';
+import { scan, takeUntil, repeat } from 'rxjs/operators';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { EventRectangle } from './content';
+
+@Injectable({ providedIn: 'root' })
+export class DragService {
+  private dragSubject = new BehaviorSubject<EventRectangleDragEvent>(null);
+  private dragEndSubject = new Subject<any>();
+
+  drag$ = this.dragSubject.pipe(
+    scan(
+      (acc, curr) =>
+        curr ? { ...curr, dy: acc.dy + curr.dy, dx: acc.dx + curr.dx } : acc,
+      {
```

Comment:
could declutter this observable/scan by declaring the initial value in its own variable ie:

```
const dragSeed = {
  dx: 0,
  dy: 0,
  id: null
}
```

Reply:
done